src/line-chart/line-chart.js

```diff
@@ -234,6 +235,47 @@ class LineChart extends AbstractChart {
     });
   };
 
+  renderLegendItems = legendItemsConfig => {
+    const { legend, data, width } = legendItemsConfig;
+    const baseLegendItemX = width / (legend.length + 1);
+
+    return legend.map((legendItem, i) => (
+      <G key={Math.random()}>
+        <LegendItem
+          index={i}
+          iconColor={this.getColor(data[i], 0.9)}
+          baseLegendItemX={baseLegendItemX}
+          legendText={legendItem}
+          labelProps={{ ...this.getPropsForLabels() }}
+          legendOffset={legendItemsConfig.legendOffset}
+        />
+      </G>
+    ));
+  };
+
+  renderLegend = (width, legendOffset, borderRadius) => {
+    const { legend, datasets } = this.props.data;
+    const legendItemsConfig = {
+      width,
+      legend,
+      legendOffset,
+      data: datasets
+    };
+
+    return (
+      <>
+        <Rect
```

Comment:
I tried removing this `Rect` and did not notice any difference, is it necessary?

Reply:
Actually, that Rect originated with a previous approach to this so I don't believe its necessary any long - will remove.

Reply:
Rect has been removed and the functionality from renderLegendItems has been moved into
renderLegend since that function was looking a little skimpy minus the Rect.